src/app/core/services/kafka.service.ts

```diff
@@ -71,7 +71,10 @@ export class KafkaService {
     // NOTE: Payload for kafka 1 : value Object which contains individual questionnaire response with timestamps
     const CompletionLog: CompletionLogValueExport = {
       name: task.name.toString(),
-      time: task.timestamp / SEC_MILLISEC,
+      time: (new Date().getTime() + Math.random()) / SEC_MILLISEC,
```

Comment:
I added this to make sure time is unique since it is used as a key for the cache.

Reply:
Hmmmm. This is fine since the max offset that will be added will be +/-1 but please document this here why this is done. Thanks.